### ISSUE TITLE
Phantom window pops up twice with autoconnect enabled

### ISSUE THREAD

**Issue by 0xCactus** [solana-labs/wallet-adapter#97]
**Describe the bug**
The popup occurs twice on a single attempt to connect

**To Reproduce**
1. Go to https://solana-labs.github.io/wallet-adapter/example/
2. Enable AutoConnect 
3. Ensure phantom extension is installed
4. Click on Dialog/Modal Button's "Select wallet" button
5. Select phantom option in dialog
6. Phantom popup show appear, close it
7. (Unexpected behaviour) Another Phantom popup appears

**Expected behavior**
Expected step 7 to not occur

**Screenshots**
https://user-images.githubusercontent.com/85321734/134789994-90d02862-cae8-4bbc-8e09-d9a5515142fd.mov

**Desktop (please complete the following information):**
 - OS: macOS big sur version 11.5
 - Browser: Brave 
 - Version: Version 1.29.80 Chromium: 93.0.4577.63 (Official Build) (x86_64)

**Smartphone (please complete the following information):**

**Additional context**

**Comment by jordansexton** [comment]
Thanks for the report, can you say what version of the library packages you're seeing this with?

**Comment by 0xCactus** [comment]
@solana/wallet-adapter-wallets@0.7.6
@solana/wallet-adapter-base@0.5.2
@solana/wallet-adapter-phantom@0.5.2
@solana/wallet-adapter-react@0.9.2
@solana/wallet-adapter-react-ui@0.1.2
 @solana/wallet-adapter-walletconnect@0.1.0

**Comment by jordansexton** [comment]
I believe this issue may be fixed in the latest versions. Can you update and tell me if it still occurs?

**Comment by jordansexton** [comment]
Also, it looks like you're trying to use the adapter for WalletConnect which has not been published to npm. This is noted on the readme and will not work.

**Comment by 0xCactus** [comment]
Sorta related to this issue but if I'm using `autoconnect` with the following logic. It also leads to multiple wallets popping up
```
onClick={async () => {
              select(wallet.name);
              await wallet
                .adapter()
                .connect()
                .catch((error: WalletError) => {
                    // do nothing
                 });
            }}
```

**Comment by jordansexton** [comment]
Thanks for the additional detail, I'll try to debug.

You shouldn't call `wallet.adapter()` yourself -- this creates another instance of the adapter. If autoconnect is also enabled, getting two connection attempts is expected behavior.

Call `connect` on the context (not the adapter) if autoconnect is disabled, and don't call anything if it's enabled.

**Comment by jordansexton** [comment]
Fixed by 01a643d3aabfefc8b76b955ce3f532e09178df47. This was introduced by 98c98d704a50a9112d374bfb76509dede714764b.
